app/controllers/api/v1/lobbies_controller.rb

```diff
@@ -59,28 +78,53 @@ def update
       end
 
       def destroy
-        @lobby = Lobby.find(params[:id])
         @lobby.destroy
 
         render :show
       end
 
       private
 
-      def require_authorisation
-        lobby = Lobby.find(params[:id])
+      def trigger_answer_sent
+        players = @lobby.players.to_a.select do |player|
+          player.player_answers.any? do |player_answer|
+            player_answer.answer.in?(@answers)
+          end
+        end
+
+        Pusher.trigger(@lobby.code, Lobby::ANSWER_SENT, { answer_count: players.count })
+      end
 
-        head :unauthorized if lobby.quiz.user != current_user
+      def create_new_answers
+        answer_params.each do |val|
+          answer = Answer.find_by(id: val)
+
+          return false if answer.nil?
```

Comment:
This will stop the answer creation mid-way if one of the answers doesn't exist.

To better handle this, depending on what you want to happen, you have two solutions:
- use `next` instead of `return false` - this would skip creating a `PlayerAnswer` for the `Answer`s that don't exist
- use a transaction (example below) and raise an exception - this would "undo" all other `PlayerAnswer` creations.

```ruby
PlayerAnswer.transaction do 
  answer_params.each do |val|
    answer = Answer.find_by(id: val)

    raise ActiveRecord::Rollback if answer.nil?

    PlayerAnswer.create(player: @player, answer: answer)
  end
end
```

Reply:
In case of a transaction, you can rely on the ActiveRecord methods that raise an exception in case things "go south" (`Answer.find` raises if it cannot find a record and `PlayerAnswer.create!` raises if the model creation fails)